web/src/reactions.ts

```diff
@@ -678,7 +678,7 @@ export function update_vote_text_on_message(message: Message): void {
     // users depends on total reactions on the message, we need to
     // recalculate this whenever adjusting reaction rendering on a
     // message.
-    set_clean_reactions(message);
+    message.message_reactions = get_message_reactions(message);
```

Comment:
I believe the intent is that we should only be accessing the cleaned reactions once we've processed them, so I'm a bit skeptical of this change.

@evykassirer can you help figure this out? Of course the context is c22a94dc89e6db3662ec7f2d2a00d35406144cbd.

Reply:
@kennethnrk note this comment; which refers to `message.reactions` but also seems very relevant.
```
    // We don't maintain message.reactions when users react to                                          
    // messages we already have a copy of, so it's safest to delete it                                  
    // after we've processed the reactions data for a message into the                                  
    // clean_reactions data structure, which we do maintain.                                            
    delete message.reactions;                                                                           
```

Reply:
> I believe the intent is that we should only be accessing the cleaned reactions once we've processed them, so I'm a bit skeptical of this change.

`message_reactions` contains the cleaned reactions, just in array form whereas `cleaned_reactions` is a Map.
If we do not intend to maintain `message_reactions` why are we rendering the reactions from it?

Reply:
Yeah it seems like we probably don't want to use a `message.message_reactions` at all. When we render, which seems to happen when we call `render_single_message`, we could do this:

```js
        const msg_reactions = reactions.get_message_reactions(message_container.msg);
        const msg_to_render = {
            ...message_container,
            message_list_id: this.list.id,
            msg_reactions,
        };
```

Reply:
Ok `message_to_render.msg.message_reactions` was used to render reactions, now we would have to render reactions from `message_to_render.msg_reactions`.

If we're not using the `message_reactions` to render the reactions, do we still need this attribute? I don't see it being used anywhere else.

Reply:
If we are not going to be updating `message_reactions` I will change #29109 to not check `message_reactions`.

Reply:
Remove the first commit from this PR and then remove `message_reactions` from the message object and make the change I suggested in my above comment. Does that solve the issue you were trying to address overall?

Reply:
> Remove the first commit from this PR and then remove `message_reactions` from the message object and make the change I suggested in my above comment. Does that solve the issue you were trying to address overall?

Yeah I think this should fix it while also simplifying the `type Message`. Thanks for the suggestion, I'll start working on it.